### PR TITLE
Firefox fix

### PR DESCRIPTION
**Pull request by TyHil**
Firefox requires requesting host permissions instead of automatically granting them.

This PR opens a prompt to grant permissions on install and every time skedge is opened without correct permissions.

### CODE REVIEW

**Comment by nl32** [comment]
If the user tries to open the extension and doesn't have host permissions, it will now open the popup and request them. 
Also added styling so the popup uses the right font and button uses `blue-dark`

File: src/background/index.ts [hunk 1]
```diff
@@ -80,14 +83,36 @@ chrome.runtime.onMessage.addListener(function (message) {
...
+    realBrowser.action.setBadgeText({ text: '!' });
+    realBrowser.action.setBadgeBackgroundColor({ color: 'green' });
   } else {
-    chrome.action.setBadgeText({ text: '' });
+    realBrowser.action.setBadgeText({ text: '' });
+  }
+});
+
+realBrowser.runtime.onInstalled.addListener(async () => {
+  console.log('hey from install listener');
```
Comment (demonlexe) [inline src/background/index.ts:98]: nit: remove console logs for a merge into main

File: src/tabs/permissions.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { neededOrigins } from '~data/config';
+
+const realBrowser = process.env.PLASMO_BROWSER === 'chrome' ? chrome : browser;
+
+function PermissionsPls() {
+  const [granted, setGranted] = useState(false);
+
+  return (
+    <div>
+      <h1 className="">Sk.edge needs permissions to run</h1>
```
Comment (demonlexe) [inline src/tabs/permissions.tsx:14]: nit: can remove className=""

File: src/popup.tsx [hunk 3]
```diff
@@ -4,6 +4,31 @@ import React from 'react';
...
+  console.log('hey from permission check');
+  const currentPermissions: { permissions: string[]; origins: string[] } =
+    await realBrowser.permissions.getAll();
+  console.log(`current permissions: ${currentPermissions.origins}`);
+  if (
+    neededOrigins.filter(
+      (origin) => !currentPermissions.origins.includes(origin),
+    ).length !== 0
+  ) {
+    console.log('opening permission request');
```
Comment (demonlexe) [inline src/popup.tsx:20]: remove console logs

**Review by demonlexe — CHANGES_REQUESTED** [review]

**Review by demonlexe — APPROVED** [review]

**Pull request merged**